metashare/api/models.py

```diff
@@ -658,6 +660,22 @@ def remove_scratch_org(self, error):
         # SCRATCH_ORG_DELETE event:
         self.delete(should_finalize=False)
 
+    def queue_refresh_org(self):
+        from .jobs import refresh_scratch_org_job
+
+        self.currently_refreshing_org = True
+        self.save()
+        self.notify_changed()
+        refresh_scratch_org_job.delay(self)
+
+    def finalize_refresh_org(self, error=None):
+        self.currently_refreshing_org = False
+        self.save()
+        if error is None:
+            self.notify_changed("SCRATCH_ORG_REFRESH")
+        else:
+            self.notify_scratch_org_error(error, "SCRATCH_ORG_REFRESH_FAILED")
```

Comment:
@wlonk What do we want to do in the error case here (e.g. when the org has been deleted a re-created, but errors running the initial flow)? I'm thinking we should actually probably handle it similar to `finalize_provision`, where we delete it from SF:

```py
if self.url:
    self.queue_delete()
else:
    self.delete()
```

If that makes sense to you, I'll go ahead and make the change.

Reply:
Back. Yes, that sounds good.